### PR TITLE
[Bugfix] Change module to work with new asana api client version

### PR DESCRIPTION
**Pull request by wzrdtales**
# Description

These are the changes needed to make the asana module work with the updated asana gem again.
# Needed user action

With the upgrade the user now needs to use _Personal Access Tokens_ instead of _API Keys_. Thus the user also needs to replace all his _API Keys_ with _Personal Access Tokens_

fixes https://github.com/gitlabhq/gitlabhq/issues/9637

### CODE REVIEW

**Comment by wzrdtales** [comment]
Whats even more useless then "the assignment" here is the message of robocop...

"app/models/project_services/asana_service.rb:75:5: W: Useless assignment to variable - asana_client."

As of http://www.rubydoc.info/github/bbatsov/rubocop/Rubocop/Cop/Lint/UselessAssignment this would be an unused variable, which this clearly is not. Either this is buggy or there is something I don't know about ruby. This could be as I'm not used to ruby, so please advice or just adjust.

File: app/models/project_services/asana_service.rb [hunk 1]
```diff
@@ -72,8 +72,8 @@ def supported_events
   def execute(data)
     return unless supported_events.include?(data[:object_kind])
 
-    Asana.configure do |client|
-      client.api_key = api_key
+    $asana_client = Asana::Client.new do |c|
```
Comment (stanhu) [inline app/models/project_services/asana_service.rb]: You may want to change this to `@asana_client`.

**Comment by stanhu** [comment]
Also see: https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/2111

That MR includes more changes that don't necessarily have to do with fixing Asana, but it does include specs that would help detect the breaking changes earlier. I think we'd want a fix that has your changes plus the basic specs.

**Comment by stanhu** [comment]
Thanks for the PR. 8d5140502e8da08fde34f02db258d938b61eb83b was merged since it had additional documentation and tests and will be shipped in GitLab 8.3.3, so closing this for now.

**Pull request closed without merging**